atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2129,53 +2126,60 @@ public void compactInternally(TableReference tableRef) {
 
     @Override
     public NodeAvailabilityStatus getNodeAvailabilityStatus() {
-        Map<InetSocketAddress, CassandraClientPoolingContainer> currentPools = clientPool.currentPools;
-        int replicationFactor = configManager.getConfig().replicationFactor();
+        if (!doesConfigReplicationFactorMatchWithCluster()) {
+            return NodeAvailabilityStatus.TERMINAL;
```

Comment:
`TERMINAL` state implies the system is in a state that cannot be recovered without manual intervention and the box should report shutdown or something. The logs will have the info as to what is the reason behind `TERMINAL`.